hyperdrive/packages/signer/sign/src/lib.rs

```diff
@@ -0,0 +1,95 @@
+use anyhow::{anyhow, Result};
+use hyperware_process_lib::net::{NetAction, NetResponse};
+use hyperware_process_lib::{await_message, call_init, Address, Message, Request, Response};
+use serde::{Deserialize, Serialize};
+
+wit_bindgen::generate!({
+    path: "target/wit",
+    world: "sign-sys-v0",
+    generate_unused_types: true,
+    additional_derives: [PartialEq, serde::Deserialize, serde::Serialize, process_macros::SerdeJsonInto],
+});
+
+#[derive(Debug, Serialize, Deserialize)]
+enum SignerRequest {
+    Sign(SignRequest),
+    Verify { from: Address, data: SignResponse },
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct SignRequest {
+    pub site: String,
+    pub time: u64,
+    pub nonce: Option<String>,
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct SignResponse {
+    pub body: SignRequest,
+    pub message: Vec<u8>,
+    pub signature: Vec<u8>,
+}
+
+call_init!(initialize);
+fn initialize(our: Address) {
+    loop {
+        let msg = await_message();
+        match msg {
+            Err(_send_error) => {
+                // ignore send errors, local-only process
+                continue;
+            }
+            Ok(Message::Request { body, .. }) => handle_request(&our, &body).unwrap_or_default(),
+            _ => continue, // ignore responses
+        }
+    }
+}
+
+fn handle_request(our: &Address, request_bytes: &[u8]) -> Result<()> {
+    let req = serde_json::from_slice::<SignerRequest>(request_bytes)?;
+    match req {
```

Comment:
A nice side benefit of moving types into WIT API is we can then do stuff like this `try_into()?`: https://github.com/hyperware-ai/kit/blob/master/src/new/templates/rust/no-ui/chat/chat/src/lib.rs#L29

Note you can always `#[derive(SerdeJsonInto)]` on your in-Rust types as well.

Unfortunately this #derive magic is only for serde_json so far, but we can easily add another derive for rmp_serde to process_macros: https://github.com/hyperware-ai/process_macros/blob/master/src/lib.rs

For now, let's just make sure all serde_json's that can be turned into try_intos or intos are try_intos or intos